apps/server/src/shared/domain/service/lesson-copy.service.spec.ts

```diff
@@ -423,5 +430,218 @@ describe('lesson copy service', () => {
+				(el) => el.type === CopyElementType.TASK && el.title === originalTask.name
+			);
+			expect(tasksStatus).toBeDefined();
+			expect(tasksStatus).toEqual(mockedTaskStatus);
+		});
+
+		it('should call taskCopyService for the linked task', () => {
+			const { originalLesson, destinationCourse, user, originalTask, taskSpy } = setup();
+
+			const copyStatus = copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			expect(taskSpy).toHaveBeenCalledWith({
+				originalTask,
+				destinationCourse,
+				destinationLesson: copyStatus.copyEntity,
+				user,
+			});
+		});
+
+		it('should call copyHelperService', () => {
+			const { originalLesson, destinationCourse, user } = setup();
+
+			copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			expect(copyHelperService.deriveStatusFromElements).toHaveBeenCalled();
+		});
+	});
+
+	describe('when mupltiple tasks are linked to the original lesson', () => {
+		const setup = () => {
+			const user = userFactory.build();
+			const originalCourse = courseFactory.build({ school: user.school, teachers: [user] });
+			const destinationCourse = courseFactory.build({ school: user.school, teachers: [user] });
+			const originalLesson = lessonFactory.build({
+				course: originalCourse,
+			});
+			const originalTasks = taskFactory.buildList(2, {
+				course: originalCourse,
+				lesson: originalLesson,
+			});
+			const taskCopyOne = taskFactory.build({ name: originalTasks[0].name });
+			const taskCopyTwo = taskFactory.build({ name: originalTasks[1].name });
+			const mockedTaskStatusOne = {
+				title: taskCopyOne.name,
+				type: CopyElementType.TASK,
+				status: CopyStatusEnum.SUCCESS,
+				copyEntity: taskCopyOne,
+			};
+			const mockedTaskStatusTwo = {
+				title: taskCopyTwo.name,
+				type: CopyElementType.TASK,
+				status: CopyStatusEnum.SUCCESS,
+				copyEntity: taskCopyTwo,
+			};
+			const mockedTaskGroupStatus = {
+				type: CopyElementType.TASK_GROUP,
+				status: CopyStatusEnum.SUCCESS,
+				elements: [mockedTaskStatusOne, mockedTaskStatusTwo],
+			};
+			jest
+				.spyOn(taskCopyService, 'copyTaskMetadata')
+				.mockReturnValueOnce(mockedTaskStatusOne)
+				.mockReturnValueOnce(mockedTaskStatusTwo);
+
+			return {
+				user,
+				destinationCourse,
+				originalLesson,
+				mockedTaskStatusOne,
+				mockedTaskStatusTwo,
+				mockedTaskGroupStatus,
+			};
+		};
+
+		it('should put copy status for each copied task under tasks', () => {
+			const { originalLesson, destinationCourse, user, mockedTaskStatusOne, mockedTaskStatusTwo } = setup();
+
+			const copyStatus = copyService.copyLesson({
+				originalLesson,
+				destinationCourse,
+				user,
+			});
+			const tasksGroupStatus = copyStatus.elements?.find((el) => el.type === CopyElementType.TASK_GROUP);
+			expect(tasksGroupStatus).toBeDefined();
+			expect(tasksGroupStatus?.elements).toEqual([mockedTaskStatusOne, mockedTaskStatusTwo]);
+		});
```

Comment:
how is it diferent from when you have one single task?

Reply:
just that it also works for more then one ^^